src/SprykerEco/Zed/PunchoutCatalogs/Presentation/_partial/connection-form.twig

```diff
@@ -21,9 +21,33 @@
     <hr>
     {{ form_row(punchoutCatalogConnectionForm.type) }}
     {% for key,choice in punchoutCatalogConnectionForm.type.vars.choices %}
-        {{ form_row(attribute(punchoutCatalogConnectionForm, choice.value)) }}
+        {{ _self.renderFormWithTemplateRecursive(attribute(punchoutCatalogConnectionForm, choice.value)) }}
```

Comment:
Isn't it better to split the logic into separate twig templates and use them in a loop, instead of going with macros? Keep It Short and Simple.

Reply:
Any idea how to do recursive render without `macros`?

Reply:
Discussed with FE-dev. Keep the current implementation.

Reply:
I discussed with @tolerants and he told me we do not need split macro into separate twig templates, because now we do not have additional logic for macro